src/cascadia/Remoting/Monarch.cpp

```diff
@@ -571,4 +572,37 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
         result->WindowName(targetWindowName);
         return *result;
     }
+
+    // Method Description:
+    // - This is an event handler for the IdentifyWindowsRequested event. A
+    //   Peasant may raise that event if they want _all_ windows to identify
+    //   themselves.
+    // - This will tell each and every peasant to identify themselves. This will
+    //   eventually propagate down to TerminalPage::IdentifyWindow.
+    // Arguments:
+    // - <unused>
+    // Return Value:
+    // - <none>
+    void Monarch::_identifyWindows(const winrt::Windows::Foundation::IInspectable& /*sender*/,
+                                   const winrt::Windows::Foundation::IInspectable& /*args*/)
+    {
+        // Notify all the peasants to display their ID.
+        for (const auto& [id, p] : _peasants)
+        {
+            try
+            {
+                p.DisplayWindowId();
+            }
+            catch (...)
+            {
+                LOG_CAUGHT_EXCEPTION();
+                // If this fails, we don't _really_ care. Just move on to the
+                // next one. Someone else will clean up the dead peasant.
+                TraceLoggingWrite(g_hRemotingProvider,
+                                  "Monarch_identifyWindows_Failed",
+                                  TraceLoggingInt64(id, "peasantID", "The ID of the peasant which we could not identify"),
+                                  TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+            }
+        }

```

Comment:
nit: This is calling out for a templated helper at this point! The only downside would be that we would use the same TraceLoggingWrite _event name_ for all of em.

```cpp
_ForAllPeasantsIgnoringTheDeid([](auto&& p, auto&& id) {
    p.DisplayWindowId();
});
```

Reply:
or 
```cpp
_ForAllPeasantsIgnoringTheDeid("Monarch_identifyWindows_Failed", 
                               [](auto&& p, auto&& id) {
    p.DisplayWindowId();
});
```